### PR TITLE
Fix Potential NULL pointer dereference

### PR DESCRIPTION
**Pull request by icy17**

### CODE REVIEW

File: test/sslapitest.c [hunk 1]
```diff
@@ -1911,6 +1911,9 @@ static int test_tlsext_status_type(void)
     /* First just do various checks getting and setting tlsext_status_type */
 
     clientssl = SSL_new(cctx);
+    if (clientssl == NULL)
+        goto end;
+
```
Comment (t8m) [inline test/sslapitest.c:1916]: This needs to be changed to `if (!TEST_ptr(clientssl))` and similarly below.

Comment (icy17) [reply test/sslapitest.c:1916]: OK, I'll fix it

**Comment by slontis** [comment]
Please read https://openssl-library.org/policies/cla/index.html

This could be classed as a trivial change..
Can you collapse this down to 1 commit via a rebase, that will also allow you to add the trivial line to the commit message

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Agreed, this would be considered trivial.

**Comment by icy17** [comment]
> This could be classed as a trivial change..
> Can you collapse this down to 1 commit via a rebase, that will also allow you to add the trivial line to the commit message

I notice that the doc says "in some cases one-line bugfixes might be accepted as trivial without requiring a CLA".  
Should I make sure my changes are within one line?

**Comment by slontis** [comment]
Not necessary, your change is simple enough.

**Review by paulidale — APPROVED** [review]

**Review by InfoHunter — APPROVED** [review]

File: test/sslapitest.c [hunk 2]
```diff
@@ -1911,6 +1911,9 @@ static int test_tlsext_status_type(void)
     /* First just do various checks getting and setting tlsext_status_type */
 
     clientssl = SSL_new(cctx);
+    if (!TEST_ptr(clientssl))
+        goto end;
+
```
Comment (nhorman) [inline test/sslapitest.c:1916]: nit: you've got a trailing whitespace character here

**Review by nhorman — CHANGES_REQUESTED** [review]

File: test/sslapitest.c (hunk 2 above)
Comment (t8m) [reply test/sslapitest.c:1916]: That can be removed when merging.

**Review by t8m — APPROVED** [review]

File: test/sslapitest.c (hunk 2 above)
Comment (icy17) [reply test/sslapitest.c:1916]: > nit: you've got a trailing whitespace character here

I'll fix it

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**